src/go/k8s/pkg/console/configmap.go

```diff
@@ -53,9 +55,18 @@ func NewConfigMap(
 
 // Ensure implements Resource interface
 func (cm *ConfigMap) Ensure(ctx context.Context) error {
-	if cm.consoleobj.Status.ConfigMapRef != nil {
+	if ref := cm.consoleobj.Status.ConfigMapRef; ref != nil {
 		cm.log.V(debugLogLevel).Info("config map ref still exist", "config map name", cm.consoleobj.Status.ConfigMapRef.Name, "config map namespace", cm.consoleobj.Status.ConfigMapRef.Namespace)
-		return nil
+		// Check ConfigMap is present, in case it is manually deleted
+		err := cm.Get(ctx, client.ObjectKey{Namespace: ref.Namespace, Name: ref.Name}, &corev1.ConfigMap{})
+		if apierrors.IsNotFound(err) {
+			cm.consoleobj.Status.ConfigMapRef = nil
+			if updateErr := cm.Status().Update(ctx, cm.consoleobj); updateErr != nil {
+				return updateErr
+			}
+			return &resources.RequeueError{Msg: err.Error()}
```

Comment:
Only on debug:
```
log.V(debugLogLevel).Info(fmt.Sprintf("Requeue ensuring resource: %s", r.Msg))
```

Reply:
I know it is on debug level. I was curious did you had a chance to see how it will look like in real environment. Maybe it lack some context as controller will error out with NotFound error while this is correct behaviour. If the error had some context attach to it, it would be much easier to debug for people that aren't aware of the implementation detail.